File: backend/models/health.py
from pydantic import BaseModel


class HealthMetric(BaseModel):
    date: str
    resting_hr: float | None = None
    hrv_avg: float | None = None
    hrv_status: str | None = None
    sleep_score: float | None = None
    sleep_duration: float | None = None
    sleep_deep: float | None = None
    sleep_rem: float | None = None
    body_battery_charged: float | None = None
    body_battery_drained: float | None = None
    stress_avg: float | None = None
    spo2_avg: float | None = None


class ReadinessScore(BaseModel):
    date: str
    score: int
    level: str          # OPTIMAL / MODERATE / LOW
    hrv_avg: float | None = None
    sleep_score: float | None = None
    body_battery: float | None = None
    stress_avg: float | None = None
    components: dict[str, float]

File: mobile/src/api/health.ts
import { useQuery } from '@tanstack/react-query';
import { getClient } from './client';

export interface HealthMetric {
  date: string;
  resting_hr: number | null;
  hrv_avg: number | null;
  sleep_score: number | null;
  body_battery_charged: number | null;
  stress_avg: number | null;
}

export interface ReadinessScore {
  date: string;
  score: number;
  level: 'OPTIMAL' | 'MODERATE' | 'LOW';
  hrv_avg: number | null;
  sleep_score: number | null;
  body_battery: number | null;
  stress_avg: number | null;
  components: Record<string, number>;
}

export function useReadiness() {
  return useQuery({
    queryKey: ['readiness', 'today'],
    queryFn: async () => {
      const { data } = await getClient().get<ReadinessScore>('/readiness/today');
      return data;
    },
    staleTime: 10 * 60_000,
  });
}

export function useHealthMetrics(days = 30) {
  return useQuery({
    queryKey: ['health-metrics', days],
    queryFn: async () => {
      const { data } = await getClient().get<HealthMetric[]>('/health-metrics', {
        params: { days },
      });
      return data;
    },
    staleTime: 10 * 60_000,
  });
}
